BLEND-181: Improve installing render delegate

VascoPi opened the pull request:
Improved Addon properties UI.
Added Delegate directory settings.
Made delegate list dynamic.

bnagirniak committed:
BLEN-120: Integrate HdRPR render delegate.
Initial.

bnagirniak committed:
Adding boost_python310-xxx.lib

bnagirniak committed:
Adjusted lib folder

bnagirniak committed:
Merge branch 'hdusd_integration' into BLEN-120

bnagirniak committed:
Continuing

bnagirniak committed:
Continuing

bnagirniak committed:
Added stage_get_info

bnagirniak committed:
Fixed using of usd_usd_ms.dll and fixed loading of HdRprPlugin from c++

bnagirniak committed:
improvements

bnagirniak committed:
Cleanups and fixes

VascoPi committed:
BLEND-181: Improve installing render delegate.
Impoved Addon properties UI.
Added Delegate directory settings.
Made delegate list dynamic.
Added Delegates folder.

VascoPi committed:
Merge branch 'hdusd_integration' into BLEN-181

VascoPi committed:
Small fix.

bnagirniak commented on extern/usdhydra/addon/properties/preferences.py:
```diff
@@ -89,6 +96,12 @@ def update_log_level(self, context):
     )
+    settings: EnumProperty(
+        name="",
+        items=(('SETTINGS', "Settings", "aaa"),
```
fix "aaa"

bnagirniak commented on extern/usdhydra/addon/utils/delegate.py:
```diff
@@ -42,16 +31,26 @@ def set_progress(self, msg):
+
+    def register_delegates(self):
+        from ..engine import init, exit
+        exit()
```
better: `from .. import engine`

bnagirniak commented on extern/usdhydra/python.cpp:
```diff
@@ -37,14 +37,10 @@ static PyObject *init_func(PyObject * /*self*/, PyObject *args)
+
   for (pxr::PlugPluginPtr p : registry.GetAllPlugins()) {
     printf("%s %s\n", p->GetName().c_str(), p->GetPath().c_str());
   }
```
remove this

bnagirniak commented on extern/usdhydra/addon/properties/preferences.py:
```diff
@@ -99,40 +112,40 @@ def _draw_settings(layout):
-        def _draw_delegate_item(layout):
-            for key, value in get_delegates().items():
-                row = layout.row(align=True)
+            plugins = Plug.Registry().GetAllPlugins()
```
remove this

VascoPi committed:
Code cleanup.
Added create_delegates_folder.
Fixed comments.

VascoPi committed:
Fix Install button.

VascoPi committed:
Code cleanup.

bnagirniak committed:
Improved working with render delegates

bnagirniak committed:
cleanups

VascoPi committed:
Code cleanup.
Removed unused module delegate.py.
Change button label "Add Render Delegate" to "Install Render Delegate".
